securedrop/manage.py

```diff
@@ -66,9 +66,13 @@ def add_journalist():  # pragma: no cover
     return _add_user()
 
 
+def get_username():  # pragma: no cover
+    return raw_input('Username: ')
+
+
 def _add_user(is_admin=False):  # pragma: no cover
     while True:
-        username = raw_input('Username: ')
+        username = get_username()
```

Comment:
The user gets an ugly traceback when they enter an invalid username:

```
vagrant@development:/vagrant/securedrop$ ./manage.py add-admin
Username: jen
Password: 
Confirm Password: 
Will this user be using a YubiKey [HOTP]? (y/N): 
['Traceback (most recent call last):\n', '  File "./manage.py", line 108, in _add_user\n    otp_secret=otp_secret)\n', '  File "<string>", line 4, in __init__\n', '  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/state.py", line 414, in _initialize_instance\n    manager.dispatch.init_failure(self, args, kwargs)\n', '  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__\n    compat.reraise(exc_type, exc_value, exc_tb)\n', '  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/state.py", line 411, in _initialize_instance\n    return manager.original_init(*mixed[1:], **kwargs)\n', '  File "/vagrant/securedrop/db.py", line 260, in __init__\n    .format(username, Journalist.MIN_PASSWORD_LEN))\n', 'InvalidUsernameException: Username "jen" must be 12 characters long.\n']
```

Two suggestions:

1. If the username is shorter than `Journalist.MIN_USERNAME_LEN`, then let's print a message like, e.g. "Username "jen" must be 8 characters long." 

2. The if block checking the username length should appear _after_ the username is input and _before_ the passwords are input. It would be annoying for a user to have to type out their desired password twice only to find their username is invalid (UX issue similar to #2035).